.github/workflows/regression-tests.yml

```diff
@@ -0,0 +1,472 @@
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: NomicFoundation/edr
+          ref: ${{ inputs.edr-ref }}
+      - uses: ./.github/actions/setup-node
+      - name: Install dependencies
+        run: pnpm install --frozen-lockfile --prefer-offline
+      - name: Download all artifacts
+        uses: actions/download-artifact@v4
+        with:
+          path: crates/edr_napi/artifacts
+      - name: Install sponge
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y moreutils
+      - name: Move artifacts
+        run: pnpm artifacts
+      - name: Install artifacts
+        run: |
+          for artifact in npm/*/*.node; do
+            pnpm install --save "./$(dirname $artifact)"
+          done
+      - name: Modify package.json
+        env:
+          JSON: |
+            {
+              "bundleDependencies": true
+            }
+        run: jq --argjson json "$JSON" '. + $json' package.json > package.json.tmp && mv package.json.tmp package.json
+      - name: Create the package
+        run: pnpm pack --config.node-linker=hoisted
+      - name: Upload the package
+        uses: actions/upload-artifact@v4
+        with:
+          name: edr
+          path: crates/edr_napi/*.tgz
+          if-no-files-found: error
+
+  build-and-publish-hardhat:
+    name: Build and publish Hardhat
+    needs: [publish-edr]
+    if: (!cancelled() && inputs.hardhat-ref != 'next' && !startsWith(inputs.hardhat-ref, '3') && inputs.edr-ref == '' || needs.publish-edr.result == 'success')
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: v-next/hardhat
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          repository: NomicFoundation/hardhat
+          ref: ${{ inputs.hardhat-ref }}
+      - uses: ./.github/actions/setup-env
+      - name: Install dependencies
+        run: |
+          pnpm install --frozen-lockfile --prefer-offline
+      - name: Download EDR
+        if: inputs.edr-ref != ''
+        uses: actions/download-artifact@v4
+        with:
+          name: edr
+          path: v-next/hardhat
+      - name: Install EDR
+        if: inputs.edr-ref != ''
+        run: pnpm install ./*.tgz
+      - name: Build
+        run: pnpm build
+      - name: Modify package.json
+        env:
+          JSON: |
+            {
+              "bundleDependencies": true
+            }
+        run: jq --argjson json "$JSON" '. + $json' package.json > package.json.tmp && mv package.json.tmp package.json
+      - name: Modify top-level package.json
+        env:
+          JSON: |
+            {
+              "pnpm": {
+                "supportedArchitectures": {
+                  "os": ["win32", "darwin", "linux"],
+                  "cpu": ["x64", "arm64"]
+                }
+              }
+            }
+        run: jq --argjson json "$JSON" '. + $json' package.json > package.json.tmp && mv package.json.tmp package.json
```

Comment:
Setting supported architectures ensures that we get all the required optional dependencies included in our Hardhat tarball.